spotinst_sdk2/clients/ocean/__init__.py

```diff
@@ -1693,3 +1694,662 @@ def list_migrations(self, ocean_id: str):
+        ocean_id (String): ID of the Ocean Cluster
+        aggregated_cluster_costs (AggregatedClusterCosts): Aggregated Cluster Costs request
+
+        # Returns
+        (Object): Aggregated Cluster Costs API response
+        """
+        aggregated_cluster_costs_request = gcp_ocean.AggregatedClusterCostRequest(
+            aggregated_cluster_costs)
+
+        excluded_missing_dict = self.exclude_missing(
+            json.loads(aggregated_cluster_costs_request.toJSON()))
+
+        formatted_missing_dict = self.convert_json_with_list_of_lists(
+            excluded_missing_dict, self.underscore_to_camel)
+
+        body_json = json.dumps(formatted_missing_dict)
+
+        aggregated_costs_response = self.send_post(
+            body=body_json,
+            url=self.__base_ocean_cluster_url + "/" + ocean_id + "/aggregatedCosts/summary",
+            entity_name='ocean (aggregated cluster costs)')
+
+        formatted_response = self.convert_json(
+            aggregated_costs_response, self.camel_to_underscore)
+
+        return formatted_response["response"]["items"][0]
+
+    def create_virtual_node_group(self, vng: gcp_ocean.VirtualNodeGroup):
+        """
+        Create a virtual node group.
+
+        # Arguments
+        vng (VirtualNodeGroup): VirtualNodeGroup Object
+        initial_nodes: When set to an integer greater than 0, a corresponding number of nodes will be launched from the virtual node group created.
+
+        # Returns
+        (Object): Ocean Launch Spec response
+        """
+        ocean = gcp_ocean.VNGRequest(vng)
+
+        excluded_missing_dict = self.exclude_missing(
+            json.loads(ocean.toJSON()))
+
+        formatted_missing_dict = self.convert_json(
+            excluded_missing_dict, self.underscore_to_camel)
+
+        body_json = json.dumps(formatted_missing_dict)
+
+        response = self.send_post(
+            body=body_json,
+            url=self.__base_ocean_launchspec_url,
+            entity_name='ocean_gcp_vng')
+
+        formatted_response = self.convert_json(response,
+                                               self.camel_to_underscore)
+
+        return formatted_response["response"]["items"][0]
+
+    def get_all_ocean_vngs(self, ocean_id: str):
+        """
+        List the configurations for all virtual node groups in the account
+        or in a specified cluster.
+
+        # Returns
+        (Object): Ocean VNG API response
+        """
+
+        response = self.send_get(
+            url=self.__base_ocean_launchspec_url,
+            entity_name="ocean_gcp_vng",
+            query_params=dict(oceanId=ocean_id)
+        )
+
+        formatted_response = self.convert_json(
+            response, self.camel_to_underscore)
+
+        return formatted_response["response"]["items"]
+
+    def import_vng_configuration(self, node_pool_name: str, ocean_id: str):
+        """
+        Import cluster configuration of an AKS cluster to use in create_ocean_cluster api call
```

Comment:
please modify the description to GKE cluster

Reply:
Done